contrib/automation_tests/test_cases/capture_window.py

```diff
@@ -338,3 +345,55 @@ def _execute(self, track_name_filter: str, expect_exists: bool = True):
                 self.expect_true(track.callstacks is not None, 'Track "%s" has callstacks pane' % track.name)
             else:
                 self.expect_true(track.callstacks is None, 'Track "%s" has no callstacks pane' % track.name)
+
+
+class SetAndCheckMemorySamplingPeriod(E2ETestCase):
```

Comment:
As we discussed offline, I am not sure about this part of the test. The part of adding it to the Capture test and verifying the existence of the track is great, but in this test, we're basically testing the behavior and correct configuration of the QLineEdit. This should IMO not happen at this level, but rather in a lower level test. It also adds a bunch of duplicated code (to show / hide the options window, and to enable the memory collection), and now that I look at it, I think it should be removed and replaced by another test if possible. @danielfenner WDYT?

At the least, it should be added in a separate commit or even PR.

Reply:
Ok, thanks for the input, then let's leave this part of the test in. I'd argue that if this was C++ code, we would probably advocate to factor this out and remove the duplication, and we should not treat our Python Code as less important when it comes to proper architecture.

I don't feel super strong about it in this case, I'm just worried that this will then "spread" as soon as we add another test that wants to also verify something in the options dialog. But I'd leave it up to @vickyliu-go4it, I'm also fine with leaving it as it is.

Reply:
Thanks for the suggestion😄!  I put the setting and validating of memory sampling period part to a separate commit, and also add a TODO to mention that we will move the capture options dialog utilities to a common base class.